### PR TITLE
chore(deps): upgrade django to 4.2 and python to 3.10

### PR DESCRIPTION
**Pull request by JohnL17**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -38,6 +38,7 @@ RUN pip install -U poetry
 
 ARG INSTALL_DEV_DEPENDENCIES=false
 COPY pyproject.toml poetry.lock $APP_HOME/
+RUN poetry lock
```
Comment (Yelinz) [inline Dockerfile:41]: should be removed

**Review by Yelinz — CHANGES_REQUESTED** [review]
also needs change in dependabot to not ignore django >=4

File: Dockerfile (hunk 1 above)
Comment (JohnL17) [reply Dockerfile]: Ah sorry I forgot to remove it. It's removed now.

**Review by Yelinz — APPROVED** [review]

**Comment by JohnL17** [comment]
@Yelinz I did something wrong when I did the rebase. So I changed that and now the tests run through again. But it would be great if you could review the changes again.

**Comment by Yelinz** [comment]
Looked good still

**Pull request merged**